_printf.c

```diff
@@ -0,0 +1,40 @@
+#include "main.h"
+
+/**
+  * _printf - function that generatess output in accordance to a given format
+  * @format: format (char, string, int, decimal)
+  * Return: size the output text;
+  */
+int _printf(const char *format, ...)
+{
+	unsigned int i, r_value = 0;
+	int (*format_char)(va_list arg);
+	va_list args;
+
+	if (format == NULL)
+		return (-1);
+
+	va_start(args, format);
+
+	for (i = 0; format[i] != 0; i++)
```

Comment:
Do you mean `for (i = 0; format[i] != '\0'; i++)` instead of `for (i = 0; format[i] != 0; i++)`

Reply:
exactly